rmf_traffic_msgs/msg/NegotiationStatus.msg

```diff
@@ -0,0 +1,23 @@
+
+# The version number assigned to this conflict
```

Comment:
these messages and topics are used by `romi-dashboard` and the `rmf_schedule_visualizer`'s websocket server

Also this dependency for `romi-dashboard`, https://github.com/osrf/romi-js-core-interfaces pulls messages from rmf_core. We could put it into rmf_schedule_visualizer but that would create more dependencies for `romi-js-core-interfaces`, so I'd rather it stay in `rmf_core` for now.

Reply:
This hasn't been documented explicitly yet, but the `rmf_traffic_msgs` package is meant for internal use by `rmf_traffic_ros2`. I don't want to commit to API or ABI stability for external use of these message definitions.

It looks like the only message definitions from `rmf_traffic_msgs` that are used in `romi-js-core-interfaces` are these two. I think that's a strong hint that these message definitions may belong elsewhere.

In particular, it's not clear to me why we would want to pack this information to a ROS2 message just to send it across SOSS and be translated into a websocket message when there aren't any other ROS2 participants subscribing to these messages. The usual motivation for doing a ROS2 --> SOSS --> Websocket translation is so that a websocket client tap into publications on a ROS2 system.

Since these messages won't have any other ROS2 listeners, I think it would make more sense to send them directly over a websocket server instead of having an unnecessary translation in the middle. At that point we won't need these message definitions anymore.

Reply:
It used to be the case whereby these messages were used by the schedule visualizer for rviz. Likewise, there was existing code being used by `romi-dashboard` to listen to ROS2 messages.

But anyway, the whole thing flows like this: 
ROS2 NegotiationStatus/Conclusion messages go to `romi-dashboard` from the server to construct/render the TreeView ui. When the user clicks on a Conflict/Table UI it requests the negotiation trajectory from the server which the server provides and gives to the dashboard upon receiving.

Some benefits of this approach are that you don't have to setup a whole new protocol for status updates, leaving me to only define the protocol for negotiation trajectories. The other is that you don't have to store as many trajectories on the frontend.

Also there is the issue that the websocket server and `romi-dashboard` was designed as a response to request model (request from romi-dashboard to websocket server), meaning there's significantly more setup to periodically listen for status update from `romi-dashboard`

Reply:
> It used to be the case whereby these messages were used by the schedule visualizer for rviz.

The rviz tool is primarily for internal development use as well, so we can afford some flexibility on that. We don't expect any operators to be using rviz in production.

> there is the issue that the websocket server and romi-dashboard was designed as a response to request model (request from romi-dashboard to websocket server), meaning there's significantly more setup to periodically listen for status update from romi-dashboard

The suggestion I'm making is to use [the existing schedule visualization websocket server](https://github.com/osrf/rmf_schedule_visualizer/blob/feature/negotiation-visualizer/rmf_schedule_visualizer/src/Server.hpp) to push these status updates instead of publishing them to a ROS2 topic and then having SOSS translate that and send it over another websocket server.

The websocket client wouldn't periodically poll for status updates; you just need to have the websocket client listen for them. This would eliminate an unnecessary ROS2 --> SOSS --> Websocket translation; we wouldn't need these message definitions anymore; and we wouldn't need any changes to `romi-js-core-interfaces`.

Either way the client needs to listen for the status updates from a websocket server, it's just that the current way requires multiple translation steps and two static message definitions that need to be compiled, while what I'm suggesting would be a more direct websocket communication and would involve two relatively simple JSON message definitions.

Reply:
I guess the gist of what I'm recommending is to merge the [`NegotiationStatusPublisher`](https://github.com/osrf/rmf_schedule_visualizer/blob/feature/negotiation-visualizer/rmf_schedule_visualizer/src/NegotiationStatusPublisher.hpp) with the [`Server`](https://github.com/osrf/rmf_schedule_visualizer/blob/feature/negotiation-visualizer/rmf_schedule_visualizer/src/Server.hpp) and do all the message passing over the visualization websocket server. I don't see any particular reason for the message passing to be fragmented the way it currently is.

Reply:
removed a4d34ef